lib/extensions/explain_validity.py

```diff
@@ -0,0 +1,154 @@
+            tspan.text = message
+        text.append(tspan)
+
+    def get_or_create_validity_layer(self):
+        svg = self.document.getroot()
+        layer = svg.find(".//*[@id='__validity_layer__']")
+
+        if layer is None:
+            layer = inkex.etree.Element(
+                SVG_GROUP_TAG,
+                {
+                    'id': '__validity_layer__',
+                    INKSCAPE_LABEL: _('Explain Validity'),
+                    INKSCAPE_GROUPMODE: 'layer',
+                    SODIPODI_INSENSITIVE: "true"
+                })
+            svg.append(layer)
+            layer = svg.find(".//*[@id='__validity_layer__']")
+
+            ignore_layer = IgnoreValidityLayer()
+            ignore_layer.insert_layer_ignore_command(layer)
+
+            text_x = str(self.unittouu(svg.get('width')) + 5)
+
+            text_container = inkex.etree.Element(
+                SVG_TEXT_TAG,
+                {
+                    "x": text_x,
+                    "y": str(5),
+                    "style": "fill:#000000;font-size:5px;line-height:1;"
+                }
+                )
+            layer.append(text_container)
+
+            text = [
+                        [_("Explain Validity"), "font-weight: bold; font-size: 6px;"],
+                        ["", ""],
+                        [_("Self-Intersection"), "font-weight: bold;"],
+                        [_("* Path > Union (Ctrl++)"), "font-size: 4px;"],
+                        [_("* Path > Break apart (Shift+Ctrl+K)"), "font-size: 4px;"],
+                        ["", ""],
+                        [_("Gap"), "font-weight: bold;"],
+                        [_("* Path > Break apart (Shift+Ctrl+K)"), "font-size: 4px;"],
+                        [_("* (Optional) Recombine shapes with holes (Ctrl+K)."), "font-size: 4px;"],
+                        ["", ""],
+                        [_("It is possible, that one object contains more than one error,"), "font-style: italic; font-size: 3px;"],
+                        [_("yet there will be only one pointer per object."), "font-style: italic; font-size: 3px;"],
+                        [_("Run this function again, when further errors occur."), "font-style: italic; font-size: 3px;"],
+                        ["", ""],
+                        [_('Remove pointers by deleting the layer named "Explain Validity"'), "font-style: italic; font-size: 3px;"],
+                        [_("through the objects panel (Object -> Objects...)."), "font-style: italic; font-size: 3px;"]
+                ]
+
+            for text_line in text:
+                tspan = inkex.etree.Element(
+                    SVG_TSPAN_TAG,
+                    {
+                        SODIPODI_ROLE: "line",
+                        "style": text_line[1]
+                    }
+                    )
+                tspan.text = text_line[0]
+                text_container.append(tspan)
+
+        return layer
+
+
+class IgnoreValidityLayer(CommandsExtension):
```

Comment:
I see what you did here, and I'm really sorry you had to go to such extreme lengths to add the "ignore layer" command!  As part of #399, I fixed this.  I made it so that `add_commands()` is no longer part of CommandsExtension, and instead it's just a function in the `commands` module that you can run anywhere.

Let's hold off on merging this pull request until we merge #399 (I swear I'll be done soon! :sweat_smile:) and then we won't need `IgnoreValidityLayer` anymore.